### PR TITLE
update parsimmon typings for parsimmon 0.9.2

### PR DESCRIPTION
**Pull request by bcherny**
- Updated definitions for Parsimmon 0.9.2
- Added Intellisense-compatible JSDoc, converted existing JSDoc

### CODE REVIEW

File: parsimmon/parsimmon.d.ts [hunk 1]
```diff
@@ -24,153 +24,233 @@ declare module 'parsimmon' {
...
+		/**
+		 * accepts a variable number of parsers that it expects to find in order, yielding an array of the results.
+		 */
+		export function seq<T>(p1: Parser<T>): Parser<[T]>;
+		export function seq<T, U>(p1: Parser<T>, p2: Parser<U>): Parser<[T, U]>;
+		export function seq<T, U, V>(p1: Parser<T>, p2: Parser<U>, p3: Parser<V>): Parser<[T, U, V]>;
+		export function seq<T, U, V, W>(p1: Parser<T>, p2: Parser<U>, p3: Parser<V>, p4: Parser<W>): Parser<[T, U, V, W]>;
+		export function seq<T, U, V, W, X>(p1: Parser<T>, p2: Parser<U>, p3: Parser<V>, p4: Parser<W>, p5: Parser<X>): Parser<[T, U, V, W, X]>;
+		export function seq<T, U, V, W, X, Y>(p1: Parser<T>, p2: Parser<U>, p3: Parser<V>, p4: Parser<W>, p5: Parser<X>, p6: Parser<Y>): Parser<[T, U, V, W, X, Y]>;
+		export function seq<T, U, V, W, X, Y, Z>(p1: Parser<T>, p2: Parser<U>, p3: Parser<V>, p4: Parser<W>, p5: Parser<X>, p6: Parser<Y>, p7: Parser<Z>): Parser<[T, U, V, W, X, Y, Z]>;
```
Comment (bcherny) [inline parsimmon/parsimmon.d.ts]: Unless someone can implement HLists in TS..

**Comment by mizunashi-mana** [comment]
@bcherny Good jobs!
I added some review comments ;)

**Comment by bcherny** [comment]
@mizunashi-mana thanks, but I don't see any comments!

**Comment by mizunashi-mana** [comment]
@bcherny Sorry, I forgot to submit reviews.  How's this?

File: parsimmon/parsimmon-tests.ts [hunk 2]
```diff
@@ -139,3 +140,31 @@ strPar = P.any;
 strPar = P.all;
 voidPar = P.eof;
 numPar = P.index;
+
+// --  --  --  --  --  --  --  --  --  --  --  --  --
```
Comment (bcherny) [inline parsimmon/parsimmon-tests.ts]: Good idea - done.

File: parsimmon/parsimmon-tests.ts [hunk 3]
```diff
@@ -139,3 +140,31 @@ strPar = P.any;
...
+
+P.isParser(numPar);
+P.isParser(null);
+P.isParser(42);
+
+P.oneOf('a');
+P.noneOf('a');
+
+P.regex(/foo/);
```
Comment (bcherny) [inline parsimmon/parsimmon-tests.ts]: Done

File: parsimmon/parsimmon.d.ts [hunk 4]
```diff
@@ -24,153 +24,233 @@ declare module 'parsimmon' {
...
+			 */
 			desc(description: string): Parser<T>;
 		}
-		/*
-		 is a parser that expects to find "my-string", and will yield the same.
+
+		/**
+		 * Returns true if obj is a Parsimmon parser, otherwise false.
+		 */
+		export function isParser<T>(obj: T): boolean;
```
Comment (bcherny) [inline parsimmon/parsimmon.d.ts]: I guess they're equivalent, though your version is simpler - done.

File: parsimmon/parsimmon.d.ts (hunk 1 above)
Comment (bcherny) [inline parsimmon/parsimmon.d.ts]: Done

**Comment by mizunashi-mana** [comment]
LGTM 👍

**Pull request merged**